### PR TITLE
feat: add start of Smithy generator integrations [WIP]

### PR DESCRIPTION
**Pull request by mtdowling**
*Issue #, if available:*

*Description of changes:*

Still a rough WIP, but this integrates the Smithy generators with AWS SDK specific plugins. Just a couple things added for now, but will continue to build this out.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: codegen/aws-typescript-models/smithy-build.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "version": "1.0",
+  "plugins": {
+    "typescript-codegen": {
+      "service": "com.amazon.rdsdataservice#RdsDataService",
+      "package": "@aws-sdk/client-rds-data-node",
```
Comment (kstich) [inline codegen/aws-typescript-models/smithy-build.json]: Can this be generated off of the `sdkId`?

File: codegen/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+
+allprojects {
+    group = "software.amazon.smithy"
+    version = "0.1.0"
+}
+
+// The root project doesn't produce a JAR.
+tasks["jar"].enabled = false
+
+// Load the Sonatype user/password for use in publishing tasks.
```
Comment (kstich) [inline codegen/build.gradle.kts]: Is only the `smithy-aws-typescript-codegen` portion going to Maven Central for now?

Comment (mtdowling) [reply codegen/build.gradle.kts]: Yes. We don't have a need to publish the models and client generation package.

File: codegen/aws-typescript-models/smithy-build.json (hunk 1 above)
Comment (mtdowling) [reply codegen/aws-typescript-models/smithy-build.json]: Yes. This file is hand written for now, with the plan of generating the file in its entirety once we've gotten Gradle iterating over the files stored in `model/`.

**Pull request merged**